app/react/kubernetes/applications/components/RedeployGitAppForm/RedeployGitAppForm.tsx

```diff
@@ -0,0 +1,362 @@
+import { Alert } from '@@/Alert';
+
+import { useUpdateKubeGitStackMutation } from './queries/useUpdateKubeGitStackMutation';
+import {
+  KubeAppGitFormValues,
+  RedeployGitStackPayload,
+  UpdateKubeGitStackPayload,
+} from './types';
+import { redeployGitAppFormValidationSchema } from './redeployGitAppFormValidationSchema';
+import { useRedeployKubeGitStackMutation } from './queries/useRedeployKubeGitStackMutation';
+
+type Props = {
+  stackId: StackId;
+  namespaceName: string;
+};
+
+export function RedeployGitAppForm({ stackId, namespaceName }: Props) {
+  const { data: stack, ...stackQuery } = useStackQuery(stackId);
+  const { trackEvent } = useAnalytics();
+  const initialValues = getInitialValues(stack);
+  const { user } = useCurrentUser();
+  const { data: gitCredentials, ...gitCredentialsQuery } = useGitCredentials(
+    user.Id
+  );
+  const environmentId = useEnvironmentId();
+  const createGitCredentialMutation = useCreateGitCredentialMutation();
+  const updateKubeGitStackMutation = useUpdateKubeGitStackMutation(
+    stack?.Id || 0,
+    environmentId
+  );
+  // keep a single generated webhook id, so that it doesn't change on every render
+  const generatedWebhookId = useRef(createWebhookId());
+  const webhookId = stack?.AutoUpdate?.Webhook || generatedWebhookId.current;
+
+  if (!initialValues) {
+    return null;
+  }
+  const isAuthEdit = !!initialValues.authentication.RepositoryUsername;
+
+  if (stackQuery.isLoading || gitCredentialsQuery.isLoading) {
+    return (
+      <InlineLoader>Loading application git configuration...</InlineLoader>
+    );
+  }
+
+  if (stackQuery.isError) {
+    return (
+      <Alert color="error">Unable to load application git configuration.</Alert>
+    );
+  }
+
+  if (!stack) {
+    return null;
+  }
+
+  return (
+    <FormSection titleSize="lg" title="Redeploy from git repository">
+      <div className="form-group text-muted">
+        <div className="col-sm-12">
+          <p>Pull the latest manifest from git and redeploy the application.</p>
+        </div>
+      </div>
+      <Formik<KubeAppGitFormValues>
+        initialValues={initialValues}
+        validationSchema={() =>
+          redeployGitAppFormValidationSchema(gitCredentials || [], isAuthEdit)
+        }
+        onSubmit={handleSubmit}
+      >
+        <RedeployGitAppInnerForm
+          stack={stack}
+          namespaceName={namespaceName}
+          webhookId={webhookId}
+        />
+      </Formik>
+    </FormSection>
+  );
+
+  async function handleSubmit(
+    values: KubeAppGitFormValues,
+    { resetForm }: FormikHelpers<KubeAppGitFormValues>
+  ) {
+    trackSubmit(values);
+
+    // save the new git credentials if the user has selected to save them
+    if (
+      values.authentication.SaveCredential &&
+      values.authentication.NewCredentialName &&
+      values.authentication.RepositoryPassword &&
+      values.authentication.RepositoryUsername
+    ) {
+      createGitCredentialMutation.mutate({
```

Comment:
you need the credentials id to create the stack. either do this action inside the createGitStack mutation or call the rest of the code inside `onSuccess` for this mutation

Reply:
Nice spot, updated in useUpdateKubeGitStackMutation